src/Stratis.Bitcoin/P2P/PeerAddressManagerBehaviour.cs

```diff
@@ -59,37 +60,46 @@ protected override void AttachCore()
             }
         }
 
-        private void AttachedPeer_MessageReceived(NetworkPeer peer, IncomingMessage message)
+        private async Task OnMessageReceivedAsync(NetworkPeer peer, IncomingMessage message)
         {
-            if ((this.Mode & PeerAddressManagerBehaviourMode.Advertise) != 0)
+            try
             {
-                if (message.Message.Payload is GetAddrPayload getaddr)
+                if ((this.Mode & PeerAddressManagerBehaviourMode.Advertise) != 0)
                 {
-                    var endPoints = this.peerAddressManager.PeerSelector.SelectPeersForGetAddrPayload(1000).Select(p => p.EndPoint).ToArray();
-                    var addressPayload = new AddrPayload(endPoints.Select(p => new NetworkAddress(p)).ToArray());
-                    peer.SendMessageVoidAsync(addressPayload);
+                    if (message.Message.Payload is GetAddrPayload getaddr)
```

Comment:
use `if (message.Message.Payload is GetAddrPayload)` instead- no need to create a `getaddr` variable

Reply:
oh yes sure, tx